### PR TITLE
Add napari-nlm recipe

### PR DESCRIPTION
**Pull request by goanpeca**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- ~[ ] If static libraries are linked in, the license of the static library is packaged.~
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by goanpeca** [comment]
We are in the process of creating conda packages for [napari plugins](https://www.napari-hub.org/) and related dependencies. Once this recipe passes all the checks, every time a new release goes into PyPI, a PR will be created automatically, with usually very minor need for changes so the process of maintaining this moving forward is small. Also the `@conda-forge/napari` team will be available to help as needed.

You can read a bit more about conda-forge at https://conda-forge.org/#contribute

---

@jaimergp, @maweigert do you agree with being added as maintainers?

**Comment by maweigert** [comment]
> @maweigert do you agree with being added as maintainers?

Yes!

**Comment by goanpeca** [comment]
This one is ready for review @conda-forge/staged-recipes :)

File: recipes/napari-nlm/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+requirements:
+  host:
+    - pip
+    - python >=3.8
+  run:
+    - gputools
+    - magicgui
+    - napari
+    - numpy
+    - pyopencl ==2022.1.5
```
Comment (jaimergp) [inline recipes/napari-nlm/meta.yaml:26]: @maweigert, is this pyopencl version the only known one to work, or is there a chance we can relax this a little? Maybe it should only be an upper or lower bound? E.g. `>=2022.1.5` or `<=2022.1.5`?

**Comment by jaimergp** [comment]
The logs mention this warning:

```
import: 'napari_nlm'
Warning: Could not find scikit-tensor which is needed for separable approximations...
If you want to compute separable approximations, please install it with
pip install scikit-tensor-py3
```

This is not currently available on conda-forge but we could add it. How important is this package? It looks like [it's not maintained and the author suggests using a replacement](https://github.com/evertrol/scikit-tensor-py3#readme).

**Comment by goanpeca** [comment]
Pinging @maweigert

**Comment by maweigert** [comment]
> Pinging @maweigert

Sorry, missed that one. 

The package above is not important for all but few specific library functions in `gputools` and thus it is safe to ingorne ignore the above warning (i.e. no action required)

**Comment by goanpeca** [comment]
Thanks @maweigert.

Regarding opencl?

Is this pyopencl version the only known one to work, or is there a chance we can relax this a little? Maybe it should only be an upper or lower bound? E.g. `>=2022.1.5` or `<=2022.1.5?

Exact pinnings can be problematic and create conflicts with other plugins.

**Comment by goanpeca** [comment]
Pinging @maweigert regarding https://github.com/conda-forge/staged-recipes/pull/19830#issuecomment-1513824248 :)

**Comment by goanpeca** [comment]
All good thanks!

**Comment by goanpeca** [comment]
This one is ready for review @conda-forge/staged-recipes :)

**Comment by goanpeca** [comment]
This one is ready for review @conda-forge/staged-recipes :)

**Comment by jakirkham** [comment]
Thanks Gonzalo! 🙏

**Pull request merged**